ovs/lib/setup.py

```diff
@@ -1056,24 +1027,73 @@ def _rollback_setup(target_client, first_node):
                     try:
                         for service in storagerouter.services:
                             service.delete()
+                        for disk in storagerouter.disks:
+                            for partition in disk.partitions:
+                                partition.delete()
+                            disk.delete()
+                        for alba_node in storagerouter.alba_nodes:
+                            alba_node.delete()
                         storagerouter.delete()
                         if len(pmachine.storagerouters) == 0:
                             pmachine.delete()
                     except Exception as ex:
                         SetupController._log(messages='Cleaning up model failed with error: {0}'.format(ex), loglevel='error')
-
                 if first_node is True:
                     try:
                         for service in ServiceTypeList.get_by_name(ServiceType.SERVICE_TYPES.ARAKOON).services:  # Externally managed Arakoon services not linked to the storagerouter
                             service.delete()
                     except Exception as ex:
                         SetupController._log(messages='Cleaning up services failed with error: {0}'.format(ex), loglevel='error')
+            if first_node is True:
+                for key in EtcdConfiguration.base_config.keys() + ['install_time', 'plugins']:
+                    try:
+                        EtcdConfiguration.delete(key='/ovs/framework/{0}'.format(key))
+                    except (EtcdKeyNotFound, EtcdConnectionFailed, KeyError):
+                        pass
 
-            for key in EtcdConfiguration.base_config.keys() + ['install_time', 'plugins', 'hosts/{0}'.format(machine_id)]:
+            try:
+                EtcdConfiguration.delete(key='/ovs/framework/{0}'.format('hosts/{0}'.format(machine_id)))
+            except (EtcdKeyNotFound, EtcdConnectionFailed, KeyError):
+                pass
+
+            #  Arakoon and ETCD must be the last services to be removed
+            for service in ['memcached', 'arakoon-ovsdb']:
```

Comment:
Yeah, but I mean that it can have a different name. See a bunch of lines below on how to retrieve the correct name.